controllers/front/expresscheckout.php

```diff
@@ -94,30 +94,30 @@ private function _expressCheckoutInitial()
 			$nvp_request .= '&L_PAYMENTREQUEST_0_NAME'.$i.'='.urlencode($product['name']).
 					'&L_PAYMENTREQUEST_0_NUMBER'.$i.'='.urlencode((int)$product['id_product']).
 					'&L_PAYMENTREQUEST_0_DESC'.$i.'='.urlencode(strip_tags(Tools::truncate($product['description_short'], 80))).
-					'&L_PAYMENTREQUEST_0_AMT'.$i.'='.urlencode((float)$product['price_wt']).
+					'&L_PAYMENTREQUEST_0_AMT'.$i.'='.urlencode(number_format($product['price_wt'],2)).
 					'&L_PAYMENTREQUEST_0_QTY'.$i.'='.urlencode((int)$product['cart_quantity']);
 					
 			$total += ((float)$product['price_wt'] * (int)$product['cart_quantity']);
 			
 			$i++;
 		}
+		$shipping = (float)$this->context->cart->getTotalShippingCost();
+		$cupon = - $total + $totalToPay - $shipping;
 
 		if ($cart_discount = $this->context->cart->getOrderTotal(false, Cart::ONLY_DISCOUNTS))
 		{
-			$nvp_request .= '&L_PAYMENTREQUEST_0_NAME'.$i.'='.urlencode($this->paypal_mx->l('Coupon')).
-					'&L_PAYMENTREQUEST_0_AMT'.$i.'='.urlencode(number_format($cart_discount,2)).
+			$nvp_request .= '&L_PAYMENTREQUEST_0_NAME'.$i.'='.urlencode($this->paypal_mx->l('Cupón')).

```

Comment:
Hi,

Okay, I see what you mean. But you should write here `Coupon` as it was before, and then translate it in Mexican (the result will appear in the file mx.php)

Regards